### PR TITLE
Compile on deploy, save us some memory

### PR DESCRIPTION
**Pull request by alloy**
I realised that MP was using babel-register in production, which is [not meant to be used in production](https://github.com/thejameskyle/babel-handbook/blob/master/translations/en/user-handbook.md#toc-babel-register).

This PR will now compiled the sources once on deploy and then use those results instead. In addition, replacing `nodemon` with `babel-watch` should lead to slightly faster restarts during development.

Some naive benchmarks that show the additional amount of required memory when compiling on the-fly. This memory is never released, so we basically constantly use an extra 90MB (in this example).

### Before

before-start

### After

after-start

### CODE REVIEW

**Comment by alloy** [comment]
(Also 35 less dependencies :tada:)

File: index.js [hunk 1]
```diff
@@ -1,6 +1,83 @@
...
+import graphqlHTTP from 'express-graphql';
+import bodyParser from 'body-parser';
+import schema from './schema';
+import loaders from './lib/loaders';
+import config from './config';
+import { info, error } from './lib/loggers';
+import auth from './lib/auth';
+import graphqlErrorHandler from './lib/graphql-error-handler';
 
-require('babel-core/register');
```
Comment (craigspaeth) [inline index.js]: 👍

File: package.json [hunk 2]
```diff
@@ -19,7 +20,8 @@
     "danger": "danger",
     "fix": "eslint . --fix",
     "promote": "heroku pipelines:promote -r staging",
-    "dump-schema": "babel-node ./scripts/dump-schema.js"
+    "dump-schema": "babel-node ./scripts/dump-schema.js",
+    "heroku-postbuild": "babel index.js config.js -s inline -d build && babel lib -s inline -d build/lib && babel schema -s inline -d build/schema"
```
Comment (craigspaeth) [inline package.json]: 😮  quite the script. I hope this doesn't trip us up if we decide to add new folders to this project.

Comment (orta) [inline package.json]: interesting, in Mitosis I just do this: https://github.com/artsy/Mitosis/blob/master/package.json#L9

**Comment by craigspaeth** [comment]
👌 looks good to me! 90 MB doesn't seem like that big of a deal if it's not growing, but happy to have this optimization.

File: package.json (hunk 2 above)
Comment (alloy) [inline package.json]: @orta That’s because you have all source in 1 directory, MP has both `lib`, `schema`, and files in the root.

**Comment by alloy** [comment]
@craigspaeth Yeah it would only further grow if new files were required, which we obviously don’t do. More importantly, this will improve future memory debugging/optimization because we don’t keep garbage/noise around.

**Pull request merged**